ggml/src/ggml-sycl/ggml-sycl.cpp

```diff
@@ -4022,13 +3979,18 @@ static bool ggml_backend_sycl_device_supports_op(ggml_backend_dev_t dev, const g
         case GGML_OP_SUB:
         case GGML_OP_MUL:
         case GGML_OP_DIV:
+            return (op->src[0]->type == GGML_TYPE_F32);
         case GGML_OP_SQR:
         case GGML_OP_SQRT:
         case GGML_OP_SIN:
         case GGML_OP_COS:
         case GGML_OP_CLAMP:
         case GGML_OP_LOG:
-            return (op->src[0]->type == GGML_TYPE_F32);
+#if defined (GGML_SYCL_F16)
```

Comment:
With this approach the SYCL backend will still instantiate the fp16 kernels when compiling the functions `ggml_sycl_op_silu` and similar even if they cannot be selected at runtime (the compiler can't know this). This is an issue for SYCL devices that don't support fp16. The easiest way I can think of is to avoid the whole `case GGML_TYPE_F16` in https://github.com/ggml-org/llama.cpp/pull/12788/files#diff-12c3f4feec96351f8e76c06420d149605170224e067b133374dbc7ead6e5d153R584 if `GGML_SYCL_F16` is not defined.